### DISCUSSION TITLE
Does it support TF and Pytorch?

### DISCUSSION THREAD

**Discussion by josephykwang** [hpcaitech/ColossalAI#1627] [Community | Q&A]
How much rewrite is required for existing TF and Pytorch models?
Are the models compatible with TF Serve, TorchServe, or Nvidia Triton?

**Comment by FrankLeeeee** [comment]
If it is a transformer model, you can directly copy the transformer layer from [Titans](https://github.com/orgs/hpcaitech/repositories). It is compatible with torch serve/nvidia triton if it runs on a single GPU. If it needs to run on multiple GPUs, you can look at [EnergonAI](https://github.com/hpcaitech/EnergonAI).

**Reply by FrankLeeeee** [reply]
This project is built upon PyTorch, thus it requires more efforts to migrate from TF to PyTorch.
